packages/loader/container-loader/src/container.ts

```diff
@@ -592,6 +593,13 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                     [DriverHeader.createNew]: {},
                 };
             }
+            // Add container package name to the request url.
+            const codeDetails = this.context.codeDetails;
+            const packageName = isFluidPackage(codeDetails.package) ? codeDetails.package.name : codeDetails.package;
```

Comment:
Also shouldn't this logic be in all resolves by default now to enable the perf scenario?

Reply:
On load path, it may not be present initially because url is resolved before the container is fetched which was the code details. On load path we are adding it when getAbsoluteUrl is called. Refer this PR. https://github.com/microsoft/FluidFramework/pull/3944

Reply:
What do you mean by all resolvers?

Reply:
Meaning, will this field be populated in the routerlicious create new driver logic for example? I guess since it's on this generic request as a url param/header it'll be in all drivers and it's SPO specific

Reply:
Currently only the odsp resolved url has this field of codehint. We need to decide if we want to add it in IFluidResolvedURl. 
Also for this PR, lets go with only odsp one as we have a deadline plus we may do a servicing release. So we want to keep it simple. We may later add it to other url resolvers. As it is just a query param, other resolvers will ignore it.